sdr_hackrf.c

```diff
@@ -0,0 +1,323 @@
+}
+
+static void show_config()
+{
+    fprintf(stderr, "freq : %ld\n", HackRF.freq);
+    fprintf(stderr, "lna_gain : %d\n", HackRF.lna_gain);
+    fprintf(stderr, "vga_gain : %d\n", HackRF.vga_gain);
+    fprintf(stderr, "samplerate : %d\n", HackRF.rate);
+    fprintf(stderr, "ppm : %d\n", HackRF.ppm);
+}
+
+bool hackRFOpen()
+{
+    if (HackRF.device) {
+        return true;
+    }
+
+    // Calculate sample rate and frequency deviation if ppm is specified
+    if (HackRF.ppm != 0) {
+        HackRF.rate = (uint32_t)((double)HackRF.rate * (1000000 - HackRF.ppm)/1000000+0.5);
+        HackRF.freq = HackRF.freq * (1000000 - HackRF.ppm)/1000000;
+    }
+
+    int status;
+
+    status = hackrf_init();
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_init failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    status = hackrf_open(&HackRF.device);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_open failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    status = hackrf_set_freq(HackRF.device, HackRF.freq);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_set_freq failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    status = hackrf_set_sample_rate(HackRF.device, HackRF.rate);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_set_sample_rate failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    status = hackrf_set_amp_enable(HackRF.device, HackRF.enable_amp);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_set_amp_enable failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    status = hackrf_set_lna_gain(HackRF.device, HackRF.lna_gain);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_set_lna_gain failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    status = hackrf_set_vga_gain(HackRF.device, HackRF.vga_gain);
+    if (status != 0) {
+        fprintf(stderr, "HackRF: hackrf_set_vga_gain failed with code %d", status);
+        hackrf_close(HackRF.device);
+        hackrf_exit();
+        return false;
+    }
+
+    show_config();
+
+    HackRF.converter = init_converter(INPUT_UC8,
+                                      Modes.sample_rate,
+                                      Modes.dc_filter,
+                                      &HackRF.converter_state);
+    if (!HackRF.converter) {
+        fprintf(stderr, "HackRF: can't initialize sample converter\n");
+        return false;
+    }
+
+    return true;
+}
+
+struct timespec thread_cpu;
+
+int handle_hackrf_samples(hackrf_transfer *transfer)
```

Comment:
This needs to become static now that we're building with `-Wmissing-declarations` (that change postdates your PR)